src/y0/algorithm/estimation.py

```diff
@@ -0,0 +1,173 @@
+    treatment: Variable,
+    outcome: Variable,
+    data: pd.DataFrame,
+    *,
+    query_type: Literal["ate", "expectation", "probability"],
+    conditions: Optional[List[Variable]] = None,
+) -> float:
+    """Estimate the causal effect of a treatment on an outcome."""
+    if query_type == "ate":
+        return estimate_ate(
+            graph=graph, treatment=treatment, outcome=outcome, data=data, conditions=conditions
+        )
+    elif query_type == "expectation":
+        raise NotImplementedError
+    elif query_type == "probability":
+        raise NotImplementedError
+    else:
+        raise TypeError
+
+
+def estimate_ate(
+    graph: NxMixedGraph,
+    treatment: Union[Variable, List[Variable]],
+    outcome: Union[Variable, List[Variable]],
+    data: pd.DataFrame,
+    *,
+    conditions: Optional[List[Variable]] = None,
+    bootstraps: int = 0,
+    alpha: float = 0.05,
+) -> float:
+    """Estimate the average treatment effect."""
+    if conditions is not None:
+        raise NotImplementedError("can not yet handle conditional queries")
+    if isinstance(treatment, list) or isinstance(outcome, list):
+        raise NotImplementedError("can not yet handle multiple treatments nor outcomes")
+    if isinstance(treatment, CounterfactualVariable) or isinstance(outcome, CounterfactualVariable):
+        raise NotImplementedError("can not yet handle counterfactual treatments nor outcomes")
+
+    ananke_graph = graph.to_admg()
+    from ananke.estimation import CausalEffect
+
+    with redirect_stdout(None):
+        # redirect stdout gets rid of the unnecessary printing from Ananke,
+        # e.g., when CausalEffect says what estimators can be used. We take
+        # care of that explicitly below
+        causal_effect = CausalEffect(ananke_graph, treatment.name, outcome.name)
+
+    # explicitly encode suggestions from Ananke
+    if is_a_fixable(graph, treatment):
+        if is_markov_blanket_shielded(graph):
+            estimator = "eff-aipw"
+        else:
+            estimator = "aipw"
+    elif is_p_fixable(graph, treatment):
+        if is_markov_blanket_shielded(graph):
+            estimator = "eff-apipw"
+        else:
+            estimator = "apipw"
+    elif is_identifiable(graph, P(outcome @ ~treatment)):
+        estimator = "anipw"
+    else:
+        raise RuntimeError("Effect can not be estimated")
+
+    return causal_effect.compute_effect(
+        data, estimator=estimator, n_bootstraps=bootstraps, alpha=alpha
+    )
+
+
+def is_markov_blanket_shielded(graph: NxMixedGraph) -> bool:
+    """Check if the ADMG is a Markov blanket shielded.
+
+    Being Markov blanket (Mb) shielded means that two vertices are non-adjacent
+    only when they are absent from each others' Markov blankets.
+
+    This code was adapted from :mod:`ananke` ananke code at:
+    https://gitlab.com/causal/ananke/-/blob/dev/ananke/graphs/admg.py?ref_type=heads#L381-403
+
+    :param graph: A NxMixedGraph
+    :returns: bool
```

Comment:
`:returns:` goes with a sentence, not a type annotation

Reply:
Got it. I will make the changes